### PR TITLE
Fix timeouts on master-side shell commands

### PR DESCRIPTION
**Pull request by p12tic**
This PR adds an reimplementation of Twisted's util.getProcessOutput family of functions with timeout functionality. This prevents a hang of master-side shell function from causing damage to the master itself. An example of hanging master-side shell command is a git pull via ssh during which the target computer is suspended. In practice this may be caused by any network interruption during ssh operations that causes the connection to not being closed or reset appropriately.

### CODE REVIEW

File: master/buildbot/test/unit/changes/test_svnpoller.py [hunk 1]
```diff
@@ -23,8 +23,10 @@
 from buildbot.changes import svnpoller
 from buildbot.process.properties import Interpolate
 from buildbot.test.util import changesource
-from buildbot.test.util import gpo
```
Comment (tardyp) [inline master/buildbot/test/unit/changes/test_svnpoller.py:26]: should we eventually remove gpo?

**Review by tardyp — APPROVED** [review]
This looks like a lot of new code. did you base on something?

File: master/buildbot/test/unit/changes/test_svnpoller.py (hunk 1 above)
Comment (p12tic) [reply master/buildbot/test/unit/changes/test_svnpoller.py:26]: Yes, there are a couple of remaining users, these will go in another PR.

**Comment by p12tic** [comment]
> This looks like a lot of new code. did you base on something?

Some code is based on our worker-side process runner, but of course I needed to rewrite almost everything to make it compatible with master's constraints.

**Pull request merged**